src/Disks/DiskDecorator.h

```diff
@@ -107,6 +107,13 @@ class DiskDecorator : public IDisk
     bool supportsChmod() const override { return delegate->supportsChmod(); }
     void chmod(const String & path, mode_t mode) override { delegate->chmod(path, mode); }
 
+    const IDisk & getNestedDisk() const
+    {
+        if (const auto * decorator = dynamic_cast<const DiskDecorator *>(delegate.get()))
+            return decorator->getNestedDisk();
```

Comment:
also check `DiskRestartProxy`, it probably requires to override this as well.

Reply:
`DiskRestartProxy` is a child of `DiskDecorator`, so it should work

Reply:
Yes, but it will require a mutex, in DiskRestartProxy delegate should be protected by mutex.